bioptim/examples/getting_started/example_implicit_dynamics.py

```diff
@@ -0,0 +1,209 @@
+
+    x_bounds = BoundsList()
+    x_bounds.add(bounds=QAndQDotBounds(biorbd_model))
+    x_bounds[0][:, [0, -1]] = 0
+    x_bounds[0][1, -1] = 3.14
+
+    # Initial guess
+    n_q = biorbd_model.nbQ()
+    n_qdot = biorbd_model.nbQdot()
+    n_qddot = biorbd_model.nbQddot()
+    n_tau = biorbd_model.nbGeneralizedTorque()
+    x_init = InitialGuess([0] * (n_q + n_qdot))
+
+    # Define control path constraint
+    # There are extra controls in implicit dynamics which are joint acceleration qddot.
+    if implicit_dynamics:
+        u_bounds = Bounds([tau_min] * n_tau + [qddot_min] * n_qddot, [tau_max] * n_tau + [qddot_max] * n_qddot)
+    else:
+        u_bounds = Bounds([tau_min] * n_tau, [tau_max] * n_tau)
+
+    u_bounds[1, :] = 0  # Prevent the model from actively rotate
+
+    if implicit_dynamics:
+        u_init = InitialGuess([0] * (n_tau + n_qddot))
+    else:
+        u_init = InitialGuess([0] * n_tau)
+
+    return OptimalControlProgram(
+        biorbd_model,
+        dynamics,
+        n_shooting,
+        final_time,
+        x_init=x_init,
+        u_init=u_init,
+        x_bounds=x_bounds,
+        u_bounds=u_bounds,
+        objective_functions=objective_functions,
+        ode_solver=ode_solver,
+        use_sx=use_sx,
+        n_threads=n_threads,
+    )
+
+
+def solve_ocp(implicit_dynamics: bool) -> OptimalControlProgram:
+    """
+    The initialization of ocp with implicit_dynamics as the only argument
+
+    Parameters
+    ----------
+    implicit_dynamics: bool
+        implicit
+    Returns
+    -------
+    The OptimalControlProgram ready to be solved
+    """
+    model_path = "models/pendulum.bioMod"
+    n_shooting = 200  # The higher it is, the closer implicit and explicit solutions are.
+    ode_solver = OdeSolver.RK2(n_integration_steps=1)
+    time = 1
+
+    # --- Prepare the ocp with implicit dynamics --- #
+    ocp = prepare_ocp(
+        biorbd_model_path=model_path,
+        final_time=time,
+        n_shooting=n_shooting,
+        ode_solver=ode_solver,
+        implicit_dynamics=implicit_dynamics,
+    )
+
+    # --- Custom Plots --- #
+    ocp.add_plot_penalty(CostType.ALL)
+
+    # --- Solve the ocp --- #
+    sol_opt = Solver.IPOPT(show_online_optim=False)
+    sol = ocp.solve(sol_opt)
+
+    return sol
+
+
+def prepare_plots(sol_implicit, sol_explicit):
+    plt.figure()
+    tau_ex = sol_explicit.controls["tau"][0, :]
+    tau_im = sol_implicit.controls["tau"][0, :]
+    plt.plot(tau_ex, label="tau in explicit dynamics")
+    plt.plot(tau_im, label="tau in implicit dynamics")
+    plt.xlabel("frames")
+    plt.ylabel("Torque (Nm)")
+    plt.legend()
+
+    plt.figure()
+    cost_ex = np.sum(sol_explicit.cost)
+    cost_im = np.sum(sol_implicit.cost)
+    plt.bar([0, 1], width=0.3, height=[cost_ex, cost_im])
+    plt.xticks([0, 1], ["explicit", "implicit"])
+    plt.ylabel(" weighted cost function")
+
+    plt.figure()
+    time_ex = np.sum(sol_explicit.real_time_to_optimize)
```

Comment:
Does the tests on examples run the main? If so, it tries to plot things.. maybe that's the test on this file that fails!

Reply:
The test "test_plot_graphs_for_implicit_constraints" works locally, it only plots at the end sol.graphs(automatically_organize=False), I tried to remove it.

Reply:
No, I meant the automatic test of examples, not the test you wrote
they are not run per se, but they are loaded

Reply:
ok.